src/dash-table/components/FilterFactory.tsx

```diff
@@ -57,13 +57,27 @@ export default class FilterFactory {
         updateColumnFilter(map, column, operator, value, setFilter);
     };
 
+    private onToggleChange = (
+        column: IColumn,
+        map: Map<string, SingleColumnSyntaxTree>,
+        operator: FilterLogicalOperator,
+        setFilter: SetFilter,
+        toggleFilterOptions: (column: IColumn) => IColumn,
+        value: any
+    ) => {
+        const newColumn = toggleFilterOptions(column);
+
+        updateColumnFilter(map, newColumn, operator, value, setFilter);
```

Comment:
Trigger a `setProps` and return the new column configuration, use the new column configuration to update the column's query (and trigger a refresh)